core/trino-spi/src/main/java/io/trino/spi/predicate/SortedRangeSet.java

```diff
@@ -842,11 +842,26 @@ public String toString(ConnectorSession session)
 
     private String formatRanges(ConnectorSession session, int limit)
     {
-        return Stream.concat(
-                IntStream.range(0, min(getRangeCount(), limit))
+        if (isNone()) {
+            return "{}";
+        }
+        if (getRangeCount() == 1) {
+            return getRangeView(0).formatRange(session);
```

Comment:
This is used in toString and should not hide the fact this is SortedRangeSet, not a Range itself.

toString should be used for human-readable representation aiding for debugging.
if you use it in DF reporting but are concerned about additional braces that would be added here, i'm concerned about soundness of using toString there at all.

Reply:
Added back the extra braces.
Technically, I'm not relying on the vanilla `toString` override of Object but `toString(ConnectorSession session, int limit)`. I could rename it to `formatDomain(ConnectorSession session, int limit)` if that helps.

Reply:
`String formatDomain` and `String toString` would have distinct names, but the names would not explain why the implementations are different, and how the code can and how it shouldn't be evolved in the future.